### DISCUSSION TITLE
【作成中】keyball61でファーム書き込み手順でpromicroが故障したと思われる原因と対策について

### DISCUSSION THREAD

**Discussion by sawa333000** [Yowkees/keyball#717] [Q&A]
はじめまして、Keyball61に惚れ込み、先日ようやく在庫回復により入手できました。
早速作り始めているところではありますが、promicro②が故障と思われる事象が発生しました。

【故障状況】
　　※結論：promicro①、promicro②が2つ故障。
　　
　promicro②単体でもPC側で「USBポートの電力サージ」とのエラーが発生して認識しません。

　掲載写真と反対側の本体へ接続したpromicro①ではキーの反応もしていたのですが、
　掲載写真側の本体へ接続したところ、promicro①も認識しなくなりました。
　おそらく掲載写真側の基板側のショートか何かで壊れたのではないかと想定しております。

【切り分け内容】
　今回掲載写真の方では、ダイオードの向きが誤っていないかダイオードが装着された状態で、
　テスターのダイオード検査モードを使い極性検査をしていたところ、
　ダイオード特有の片側のみへの電流が流れる状態ではない箇所が2箇所ありました。

　極性を変えて当ててみても、同等のv数表示となったため、ダイオード故障かと思い、
　ダイオード自体を新しいものに交換してみましたが状況が変わらないため、
　一度基盤すべてのダイオードを取り外した状態としました。

　この状態で再度、ダイオード検査モードでテスターを当ててみたところ、
　該当2個所以外の場所では.OLという表示であるのにもかかわらず、
　該当2箇所のみ1V未満ではあるものの通電する状況でした。

【ご質問】
　以下の状態に関しましては、正常な状態でしょうか。
　　◎数字27の下のダイオードと数字10,15の下のダイオードが1V未満の通電がある。
　　　（ダイオードなし状態、それ以外は手順どおり装着済み）

　もし、異常な状況でしたら想定原因・改善方法などご教示いただけますと助かります。

この問題をクリアしてから再度新品のpromicroを購入して前に進もうと思います。
（どうしてもこのキーボードを作り上げたいので恐れ入りますがお力をおかしください。）

![IMG_9040](https://github.com/user-attachments/assets/0e8f2572-a883-4212-9960-bfa050f9456e)

**Comment by Yowkees** [comment]
詳細なご調査をしていただき、誠にありがとうございます。

ダイオードやProMicroをすべて外した状態で、赤丸の両端子に反応があるのはおかしいです。
基板が傷ついたためのショートの可能性もありますが、基板の正常不良の可能性もあります。
すでにProMicroが故障してしまったということもありますので、今後の進め方をご相談させていただきたいと思います。
お手数ですが下記の白銀ラボお問い合わせフォームよりご連絡いただくことは可能でしょうか。
https://shirogane-lab.net/inquiry

以上、お手数ですがよろしくお願いいたします。

**Comment by sawa333000** [comment]
Yowkees様

本件ご連絡いただきましてありがとうございました。
その後新たな基盤・部品を使って同様に制作を進めておりましたが、
キースイッチ、ダイオード、プロマイクロ等を装着し、ファーム書き込みを行いまして、
キーの入力について確認をしておりました。

（１）確認結果ですが２４，２８，３２のキーが無反応状態でした。
　　　（a.プロマイクロの上のディスプレイでも無反応、b.キースイッチを外してみるも端子曲がりなどはない状態）

（２）テスターのダイオードの検査モードで検査を行っていたところ、
　　　①のダイオードが、両側から電流が流れる状態が確認取れました。
　　　（c.外したダイオード単体でダイオードの検査をすると、片側しか電流を流さない状態で正常
　　　　d.ダイオードを外した基板側を測定すると逆側で電流が流れることを確認）

（３）一度、①のダイオードを外した状態で再確認をしたところ、２４と３２のキーは入力できる状態でした。

（４）改めて、①のダイオードを装着すると、２４，２８，３２のキーは再度無反応状態となります。
　　　※（１）の状態に戻りました。

　　e.その他ダイオードの極性確認は問題ありませんでした。
　　f.キースイッチのピンソケットは全体的にテンプラが無いか確認・修正しています。
　　　
この場合考えられる原因等ございますでしょうか。
何度もお手数をおかけしまして申し訳ありませんが、
　
※類似事例「keyball61一部のキーが反応しない #404」も確認させていただき
KiCADを使って回路図も見てみましたが、素人にはちょっと難しく、御指南いただけると幸いでございます。

![keyball61_27](https://github.com/user-attachments/assets/0a827389-76e8-4890-a3f3-5cc425efa536)
